backend/shared/src/create-cash-contract.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  createSupabaseDirectClient,
+  SupabaseDirectClient,
+} from './supabase/init'
+import { getContract, getUser, htmlToRichText, log } from './utils'
+import { runTxnFromBank } from './txn/run-txn'
+import { APIError } from 'common/api/utils'
+import { updateContract } from './supabase/contracts'
+import { randomString } from 'common/util/random'
+import { getNewContract } from 'common/new-contract'
+import { convertContract } from 'common/supabase/contracts'
+
+// cribbed from backend/api/src/create-market.ts
+
+export async function createCashContract(
+  manaContractId: string,
+  subsidyAmount: number,
+  pg?: SupabaseDirectClient
+) {
+  const client = pg || createSupabaseDirectClient()
+
+  return await client.tx(async (tx) => {
+    const manaContract = await getContract(tx, manaContractId)
+    if (!manaContract) {
+      throw new APIError(404, `Mana contract ${manaContractId} not found`)
+    }
+
+    const creator = await getUser(manaContract.creatorId)
+
+    if (!creator) {
+      throw new APIError(404, `Creator ${manaContract.creatorId} not found`)
+    }
+
+    if (manaContract.token !== 'MANA') {
+      throw new APIError(
+        400,
+        `Contract ${manaContractId} is not a MANA contract`
+      )
+    }
+
+    if (manaContract.outcomeType !== 'BINARY') {
+      throw new APIError(
+        400,
+        `Contract ${manaContractId} is not a binary contract`
+      )
+
+      // TODO: Add support for multi
+    }
+
+    const contract = getNewContract({
+      id: randomString(),
+      ante: subsidyAmount,
+      token: 'CASH',
+      description: htmlToRichText('<p></p>'),
+      initialProb: manaContract.prob,
+
+      creator,
+      slug: manaContract.slug,
```

Comment:
Probably good to make this unique as other code assumes so. Can add as suffix '-cash'